### PR TITLE
Widgets content classes were created.

### PR DESCRIPTION
**Pull request by Yumfriez**
Statistics queries were added to WidgetRepository

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java [hunk 1]
```diff
@@ -146,4 +161,589 @@
...
+	}
+
+	@Override
+	public List<ComparisonStatisticsContent> launchesComparisonStatistics(Filter filter, Map<String, List<String>> contentFields,
+			String launchName, int limit) {
+
+		Select commonSelect = dsl.select(field(name(LAUNCHES, "id")).cast(Long.class))
+				.distinctOn(field(name(LAUNCHES, "id")).cast(Long.class))
+				.from(name(LAUNCHES))
+				.where(field(name(LAUNCHES, "launch_name")).eq(launchName))
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java]: move to filter condition

File: src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java [hunk 2]
```diff
@@ -146,4 +161,589 @@
...
+					resultComparisonStatisticsContent.add(content);
+				}));
+
+		return resultComparisonStatisticsContent;
+	}
+
+	@Override
+	public List<LaunchesDurationContent> launchesDurationStatistics(Filter filter, Map<String, List<String>> contentFields, int limit) {
+
+		Select commonSelect = dsl.select(field(name(LAUNCHES, "id")).cast(Long.class))
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java]: move

File: src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java [hunk 3]
```diff
@@ -146,4 +161,589 @@
...
+				.distinctOn(field(name(LAUNCHES, "id")).cast(Long.class))
+				.from(name(LAUNCHES))
+				.limit(limit);
+
+		List<LaunchesDurationContent> launchesDurationContents = dsl.with(LAUNCHES)
+				.as(QueryBuilder.newBuilder(filter).build())
+				.select(LAUNCH.ID.as("launchId"),
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java]: const

File: src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java [hunk 4]
```diff
@@ -146,4 +161,589 @@
...
+				.limit(limit);
+
+		return dsl.with(LAUNCHES)
+				.as(QueryBuilder.newBuilder(filter).build())
+				.select(LAUNCH.ID.as("launchId"),
+						LAUNCH.NUMBER.as("number"),
+						LAUNCH.NAME.as("name"),
+						LAUNCH.START_TIME.as("startTime"),
+						(sum(when(EXECUTION_STATISTICS.ES_STATUS.equal(JStatusEnum.PASSED.getLiteral()),
+								0
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepositoryImpl.java]: give it some name, make it local variable for example

File: src/main/java/com/epam/ta/reportportal/entity/widget/content/AbstractLaunchStatisticsContent.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		return startTime;
+	}
+
+	public void setStartTime(Timestamp startTime) {
+		this.startTime = startTime;
+	}
+
+	@Override
+	public String toString() {
+		return MoreObjects.toStringHelper(this)
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/entity/widget/content/AbstractLaunchStatisticsContent.java]: lets use standard toString method

File: src/main/java/com/epam/ta/reportportal/entity/widget/content/UniqueBugContent.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	@JsonProperty(value = "submitter")
+	private String submitter;
+
+	@JsonProperty(value = "item_name")
+	private String testItemName;
+
+	@JsonProperty(value = "url")
+	private String url;
+
+	@JsonProperty(value = "submit date")
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/entity/widget/content/UniqueBugContent.java]: ...

File: src/main/java/com/epam/ta/reportportal/entity/widget/content/UniqueBugContent.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+/**
+ * @author Ivan Budayeu
+ */
+public class UniqueBugContent implements Serializable {
+
+	@JsonProperty(value = "item_id")
+	private Long testItemId;
+
+	@JsonIgnore
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/entity/widget/content/UniqueBugContent.java]: check this

File: src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java [hunk 8]
```diff
@@ -31,6 +32,7 @@
 
 	LAUNCH(Launch.class, Arrays.asList(
 			//@formatter:off
+			new CriteriaHolder("id", "l.id", Long.class, false),
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java]: why not constant as other params?

File: src/main/java/com/epam/ta/reportportal/dao/LaunchRepositoryCustomImpl.java [hunk 9]
```diff
@@ -55,7 +56,7 @@
 @Repository
 public class LaunchRepositoryCustomImpl implements LaunchRepositoryCustom {
 
-	private static final RecordMapper<? super Record, IssueStatistics> ISSUE_STATISTICS_RECORD_MAPPER = r -> {
+	public static final RecordMapper<? super Record, IssueStatistics> ISSUE_STATISTICS_RECORD_MAPPER = r -> {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/dao/LaunchRepositoryCustomImpl.java]: shouldn't be public

File: src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepository.java [hunk 10]
```diff
@@ -39,4 +39,134 @@
...
+
+	/**
+	 * Launch statistics content loading
+	 *
+	 * @param filter        Filter
+	 * @param contentFields Fields with restrictions
+	 * @param limit         Results limit
+	 * @return List of {@link Launch}
+	 */
+	List<Launch> launchStatistics(Filter filter, Map<String, List<String>> contentFields, int limit);
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepository.java]: i would suggest extracting xxxStatistics to separate LaunchStatisticsRepository class. Do you think it makes sense?

File: src/main/java/com/epam/ta/reportportal/entity/widget/content/FlakyCasesTableContent.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
+package com.epam.ta.reportportal.entity.widget.content;
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/entity/widget/content/FlakyCasesTableContent.java]: do not forget about file headers. apache 2.0

File: src/main/java/com/epam/ta/reportportal/entity/widget/content/NotPassedCasesContent.java [hunk 12]
```diff
@@ -0,0 +1,23 @@
+package com.epam.ta.reportportal.entity.widget.content;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * @author Ivan Budayeu
+ */
+public class NotPassedCasesContent extends AbstractLaunchStatisticsContent {
+
+	@JsonProperty(value = "% (Failed+Skipped)/Total")
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/entity/widget/content/NotPassedCasesContent.java]: not sure it's a good name for json field

**Comment by avarabyeu** [comment]
Should we merge it? or close?

**Comment by Yumfriez** [comment]
I can just close it for a while, cause now we are working on the new widgets' logic. 
And if something is wrong, we will use this branch as back up

File: src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java (hunk 8 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/commons/querygen/FilterTarget.java]: will be fixed

File: src/main/java/com/epam/ta/reportportal/dao/LaunchRepositoryCustomImpl.java (hunk 9 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/dao/LaunchRepositoryCustomImpl.java]: ok, it was a temporary measure

File: src/main/java/com/epam/ta/reportportal/entity/widget/content/NotPassedCasesContent.java (hunk 12 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/entity/widget/content/NotPassedCasesContent.java]: It's like in the 4.3 version and our front-end team requested that name of field, cause it's more convinient for them

File: src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepository.java (hunk 10 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/dao/WidgetContentRepository.java]: It could be done in the future.

**Pull request closed without merging**